### PR TITLE
fix(flow): Arrange get range with batch unaligned

### PR DESCRIPTION
**Pull request by discord9**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?
Fix arrangement's batches's interaction with get by range unaligned with batch boundaries. Related tests are added and commented in details.
## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/flow/src/utils.rs [hunk 1]
```diff
@@ -176,15 +183,85 @@ impl Arrangement {
...
+                            *ts,
+                            *diff,
+                        ));
+                    }
+                    *ts > *now
+                });
+                !updates.is_empty()
+            });
+
+            before.insert(*now, new_batch);
```
Comment (zhongzc) [inline src/flow/src/utils.rs:251]: how about `entry(*now).or_default().extend()`?

File: src/flow/src/utils.rs [hunk 2]
```diff
@@ -176,15 +183,85 @@ impl Arrangement {
...
+    /// find out the time of next update in the future
+    pub fn get_next_update_time(&self) -> Option<Timestamp> {
+        let next_batch = {
+            let mut iter = self.spine.iter();
+            if self.full_arrangement {
+                iter.next();
+                iter
+            } else {
+                iter
+            }
```
Comment (zhongzc) [inline src/flow/src/utils.rs:195]: can you add a comment to explain why?

**Comment by zhongzc** [comment]
rest LGTM

File: src/flow/src/utils.rs (hunk 2 above)
Comment (discord9) [reply src/flow/src/utils.rs]: > can you add a comment to explain why?

refactor a bit, now require passing a `now` timestamp

**Review by zhongzc — APPROVED** [review]

**Review by killme2008 — APPROVED** [review]
LGTM

**Comment by waynexia** [comment]
Forget to send them...

**Pull request merged**